### DISCUSSION TITLE
Remotecontrol_Power_Control = Enabled Feedin Priority is not working

### DISCUSSION THREAD

**Discussion by Jan1Scholz** [wills106/homeassistant-solax-modbus#403] [General]
Hallo, 
at 6am I test current spot price of electricity and if price is bellow certain threshold, I emulate Feed in Priority for 6 hours (Inverter is in self-use mode):
 __trigger:
  - platform: time
    at: 06:00:00
  condition:
  - condition: and
    conditions:
    - condition: numeric_state
      entity_id: sensor.energy_production_today
      above: 7
    - condition: numeric_state
      entity_id: sensor.spot_cheapest_electricity_today
      below: 70
  action:
  - device_id: ef46974863bd9a7da9387f0857cef629
    domain: select
    entity_id: select.solax_remotecontrol_power_control
    type: select_option
    option: Enabled Feedin Priority
  - device_id: ef46974863bd9a7da9387f0857cef629
    domain: number
    entity_id: number.solax_remotecontrol_autorepeat_duration
    type: set_value
    value: 21600
  - device_id: ef46974863bd9a7da9387f0857cef629
    domain: button
    entity_id: button.solax_remotecontrol_trigger
    type: press
  mode: single__

The emulation of Feedin Priority  **is not working** (in reality the battery is charged) and monitored solar production in HA is strange (stable for 6 hours, but standard solar production in reality) :
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/131488177/912427fd-ae4b-4a52-9ab6-faacf4b42664)
I have no idea, what is wrong ?
Thank you,
Jan

**Comment by infradom** [comment]
Thanks for using the remotecontrol features. I have been using it for a long time, and have not encountered any problems. I assume you have a solax hybrid Gen4 system.  Can you explain what the attachted graph plots? I am missing  a title. is it the PV solar power? or something else?
Make sure export_control_user_limit is set to a high value.
I must admit that the flat line for 6 hours is very weird, but as I explained, it is unclear what parameter is plotted.


In my spring/summer automations, I typically start the day with an optional discharge, and then go to emulated feedin priority till around 13:00, then go to normal selfuse mode. My feedin priority automation looks like (evaluated every hour):

```alias: Feedin Priority
description: ""
trigger:
  - platform: time_pattern
    minutes: "0"
    seconds: "15"
condition:
  - condition: numeric_state
    entity_id: sensor.solcast_forecast_remaining_today
    enabled: true
    above: 12
  - condition: template
    value_template: >-
      {{states('sensor.ecopower_injection_price') >
      states('input_number.feedin_prio_price_threshold')}}
  - condition: template
    value_template: >-
      {{states('sensor.ecopower_injection_price') <=
      states('input_number.discharge_price_threshold')}}
  - condition: template
    value_template: >-
      {{states('sensor.solax_battery_capacity') | float(10.0) >=
      states('input_number.discharge_target_soc_end') | float(0) - 2 }} 
  - condition: state
    entity_id: input_boolean.feedin_enabled
    state: "on"
action:
  - service: number.set_value
    data:
      value: "0"
    target:
      entity_id: number.solax_remotecontrol_active_power
  - service: select.select_option
    data:
      option: Enabled Feedin Priority
    target:
      entity_id: select.solax_remotecontrol_power_control
  - service: number.set_value
    data:
      value: "3590"
    target:
      entity_id: number.solax_remotecontrol_autorepeat_duration
  - service: button.press
    data: {}
    target:
      entity_id: button.solax_remotecontrol_trigger
mode: single
```

It should not matter, but I do set the remotecontrol_active_power to zero in this script

See also wiki page about modbus remotecontrol for more background info

**Reply by Jan1Scholz** [reply]
Thank you very much for your answer and excellent integration of Solax Inverter.
Yes, I use Solax hybrid Gen4 system and the attached graph  is the PV solar power. The export_control_user_limit is set to 20 000 W, which is highly above my PV capacity.  The feed-in emulation had worked for 14 days, I only updated your integration and prolonged autorepeat duration from 5 to 6 hours. 
I will try to shorten evaluation to 2 hours and to set remotecontrol_active_power to zero.
Thank you and kind regards from the Czech Republic.
Jan

**Reply by infradom** [reply]
Thanks for the info; I still cannot explain what you experience.
I looked at the code again, and it should not be necessary to set remotecontrol_active_power to zero.
I often work with long autorepeat durations (several hours) as my automations terminate the autorepeat/remotecontrol_mode when price is low

**Comment by Jan1Scholz** [comment]
Hi, it works :-) with shortening  evaluation to 2 hours and setting remotecontrol_active_power to zero. Thank you. Jan

**Comment by infradom** [comment]
Thanks for the feedback. After looking into the source code, I still annot believe that the measures you took change anything.
One note: the remotecontrol_active_power mechanism limits the active power according to the dynamic  max en min limits received from the Inverter. These limits vary constantly e.g. when the Inverter judges that batteries need some kind of rebalancing, or when batteries are nearly full.
This can explain some temporary strange behavior. It should not limit the PV power however, so I still wonder whether your plot really shows the PV power and not another parameter ...
One other explanation might be that the voltage on the grid was getting too high, and that the inverter needs to limit the PV power for that reason.

**Comment by Jan1Scholz** [comment]
Hallo. 
I have been using remotecontrol features without any problem for 14 days (every day I emulate  feedin priority from 6am to 12am). Today around 9:30 (when feedin priority was emulated), I wanted to change one parameter of my invertor via Solax app and the above mentioned error appeared again: PV production in HA was stationary, but in the reality PV production was normal; emulation of feed-in was stuck (autorepeat remaining was steady) and in the reality self-use mode was active .... I had to restart the dongle .... 
It is probably communication error and it appears when remotecontrol is working and I want to change parameter of my invertor via Solax app ....
Kind regards,
Jan

**Reply by wills106** [reply]
Are you a PocketWiFi v1 or v2 by any chance?

**Reply by Jan1Scholz** [reply]
SolaX POCKET WIFI 3.0 Black with fw 3.001.02

**Reply by wills106** [reply]
I would email SolaX and ask them to update the firmware then. The Pocket WiFi 3.0 only officially supports Modbus on V3.004.03 and above.
